src/components/Modal/Modal.module.css

```diff
@@ -0,0 +1,140 @@
+@import '../../design-tokens/mixins.css';
+
+/*------------------------------------*\
+    # MODAL
+\*------------------------------------*/
+
+/**
+ * The modal wrapper and overlay which takes up the entire screen.
+ */
+.modal,
+.modal__overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  height: 100vh;
+  width: 100%;
+}
+
+/**
+ * The inverted background of the modal to provide contrast with the actual modal.
+ */
+.modal__overlay {
+  background-color: var(--eds-theme-color-body-background-inverted);
+  opacity: 0.5;
+}
+
+/**
+ * The modal container which positions the modal in the center of the screen.
+ * 1) Set a default z-index on the modal container. This is our smallest z-index token.
+ */
+.modal {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding: var(--eds-size-2);
+
+  z-index: var(--eds-z-index-100); /* 1 */
+}
+
+/**
+ * Modal transition animations.
+ */
+.modal__transition--enter {
+  transition: opacity var(--eds-anim-fade-long) var(--eds-anim-ease);
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+.modal__transition--enterFrom {
+  opacity: 0;
+}
+.modal__transition--enterTo {
+  opacity: 1;
+}
+.modal__transition--leave {
+  transition: opacity var(--eds-anim-fade-long) var(--eds-anim-ease);
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+.modal__transition--leaveFrom {
+  opacity: 1;
+}
+.modal__transition--leaveTo {
+  opacity: 0;
+}
+
+/**
+ * The content of the modal, which can wrap header, body, and footer.
+ * 1) This transparent border is for Window High Contrast Mode, which removes all
+    background colors but makes borders 100% opacity black. Without this, the
+    modal would have no clear boundary.
+ *  2) We use flexbox to handle height and width automatically. 
+    The body can scroll when content goes beyond the component,
+    while the footer remains static (if scrollable).
+ */
+.modal__content {
+  position: relative;
+  height: 43.125rem;
+  overflow: hidden;
+  border-radius: var(--eds-border-radius-lg);
+  background-color: var(--eds-theme-color-background-neutral-default);
+  /* 1 */
+  border: var(--eds-theme-form-border-width) transparent
+    var(--eds-theme-color-background-neutral-default);
+  /* 2 */
+  display: flex;
+  flex-direction: column;
+
+  width: 22.5rem;
+  --modal-horizontal-padding: var(--eds-size-4);
+  @media all and (min-width: $eds-bp-md) {
+    width: 42rem;
+    --modal-horizontal-padding: var(--eds-size-6);
+  }
+  @media all and (min-width: $eds-bp-xl) {
+    width: 64rem;
+    --modal-horizontal-padding: var(--eds-size-8);
+  }
+}
+
+/**
+ * Allows scrolling of the modal content except for sticky footer.
+ * This functionality is our intended scroll behavior but consuming teams can 
+ * style the body content as they wish to handle overflow in various ways.
+ */
+.modal__content--scrollable {
+  overflow: scroll;
```

Comment:
Does this work? I tried putting a bunch of content in a modal story and I can't get it to scroll.

Reply:
Ah, I see, you need to pass a prop to get it to scroll.

I'm a bit concerned about that because some modals will have enough content to need to be scrollable on mobile but not enough content to be scrollable on desktop. So devs won't think to add the prop and it will be unusable on mobile...

Reply:
I definitely agree with your sentiment and spoke with the designer about this. But he advised that there will be notes to do mobile design first to handle overflow and then pass the prop as necessary. Also body content itself can be `overflow: scroll`.

One alternative is also to attach the prop based on overflow content, but that would make the component heavy

Reply:
So passing the prop will be based on whether or not the content is scrollable at the mobile level? If so, maybe add that to the prop comment.

Reply:
That makes sense to me, will add to prop comment

Reply:
I've also thought in the past about trying to do something fancy here with waiting for the component to render then comparing the client height of the body container with the scroll height, then automatically add the scroll css and tab index to the body container if the scroll height is taller than the client height. I never pursued that idea because I was afraid it might be fragile with the timing of the measurement logic, but you could consider trying that out if you're interested.

Reply:
I've had to do work like that previously and I don't really think it's worth supporting due to high complexity and possible race conditions

Reply:
Yeah that's what I was thinking; thanks for verifying! 👍

Reply:
Can we change this from `overflow: scroll;` to `overflow: auto;`? That way the scroll bars will only appear when needed (on operating systems where they appear on all scrollable content – Mac is sneaky and hides it as if the style was `overflow: auto;`). See this PR for more details: https://github.com/FB-PLP/traject/pull/10197

Reply:
Yup sounds good to me